docs/source/tutorial.rst

```diff
@@ -76,12 +76,36 @@ should be output by the client can be set by specifying the log level as shown i
 The log is printed to the standard output, normally the terminal window from which the compiler is invoked. The available log levels are 
 'INFO'(default), 'ERROR', 'WARNING', 'CRITICAL' and 'DEBUG'.
 
-Compile model
--------------
+Analysis
+--------
 
-Next, we can compile the model to an FMU for further analysis by calling the ``compile()`` method on the ``model``.
+The model analysis can be setup in one of two workflows described below:
+
+   1. **FMU ID based workflow**: This workflow requires the user to compile the model to a model executable 
+   before setting up an experiment for it. This workflow is useful when the user intends to do all the 
+   computations with the FMUs in a notebook environment(i.e., no experimentation/computation in the Modelon 
+   IMPACT server). 
+
+   **Note:** Since the user works with the compiled FMUs in the workflow, modifiers added during the
+   experimentation step should contain only non-structural parameter modifiers. Any non-structural parameter change
+   would require a recompilation of the model.
+
+   2. **Class name based workflow**: This workflow allows the user to directly work with the model class object to setup
+   experiments for analysis. The compilation in this workflow is handled as a part of the execution step.
+   This workflow allows the users to skip the details of the compilation steps and focus on setting up the experiment
+   of interest. Also, this workflow doesn't impose any restrictions on the parameter modifiers that could be 
+   applied to the model. If a structural parameter is set as a modifier in a batch run, a recompilation is triggered automatically
+   by the client. 
+
+1. FMU ID based workflow
+########################
+
+1.1 Compiling the model
+***********************
+
+We can compile the model to an FMU for further analysis by calling the ``compile()`` method on the ``model``.
 The ``compile()`` method takes 1 mandatory argument(``compiler_options``) and 6 optional(``runtime_options``, ``compiler_log_level``, 
-``fmi_target``, ``fmi_version``, ``platform``) ones. 
+``fmi_target``, ``fmi_version``, ``platform``) ones.
```

Comment:
It says 6 optional and then only lists 5. We should add `force_compilation` to this list.

Reply:
Good catch!